include/oneapi/dpl/experimental/kt/esimd_radix_sort_onesweep.h

```diff
@@ -594,6 +594,21 @@ struct __radix_sort_onesweep_submitter<_IsAscending, _RadixBits, _DataPerWorkIte
     }
 };
 
+struct __radix_sort_onesweep_mem_free_submitter
+{
+    sycl::event
+    operator()(sycl::queue& __q, void* __ptr, const sycl::event& __e) const
+    {
+        return __q.submit(
+            [&](sycl::handler& __cgh)
+            {
+                __cgh.depends_on(__e);
+                __cgh.host_task([__q, __ptr]() { sycl::free(__ptr, __q); });
```

Comment:
Usually this solution works OK excepting the case when exception occurred from some previous Kernel code - in this case we have memory leak.

Reply:
As for me this memory leak in exception case seems not important.